File: cca_core/web/ng-app/src/app/core/session/selection/partner.ts
import { Permission } from "../../auth/permission";
import { PlatformType } from "../../platform/platform-type.enum";

export class Partner {

  public static readonly INCOMM = 'INCOMM';
  public static readonly GLOBAL = 'GLOBAL';

  id: number;
  ivrDnis: string;
  name: string;
  permission: Permission;
  platform: PlatformType;
  type: string;

  constructor ( data: any ) {
    if ( data ) {
      Object.assign ( this, data );

      if ( data.permission ) {
        this.permission = new Permission ( data.permission );
      }
      if ( data.platform ) {
        this.platform = PlatformType[ <string>data.platform ];
      }
    }
  }
}

File: cca_core/web/src/main/java/com/incomm/cca/model/domain/Partner.java
package com.incomm.cca.model.domain;

import com.incomm.cca.model.domain.auth.Permission;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.OneToOne;
import javax.persistence.Table;
import java.io.Serializable;

@Entity
@Table(name = "partner")
public class Partner implements Serializable {

    private Long id;
    private String ivrDnis;
    private String platform;
    private String name;
    private Permission permission;
    private String type;

    public void setId(Long id) {
        this.id = id;
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getIvrDnis() {
        return ivrDnis;
    }

    public void setIvrDnis(String ivrDnis) {
        this.ivrDnis = ivrDnis;
    }

    public String getPlatform() {
        return platform;
    }

    public void setPlatform(final String platform) {
        this.platform = platform;
    }

    @OneToOne(cascade = {CascadeType.PERSIST, CascadeType.REMOVE})
    @JoinColumn(name = "permission_id")
    public Permission getPermission() {
        return this.permission;
    }

    public void setPermission(Permission permission) {
        this.permission = permission;
    }

    public String getType() {
        return type;
    }

    public void setType(final String type) {
        this.type = type;
    }
}
